src/Servers/Connections.Abstractions/src/NamedPipeEndPoint.cs

```diff
@@ -0,0 +1,65 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics.CodeAnalysis;
+using System.Net;
+
+namespace Microsoft.AspNetCore.Connections;
+
+/// <summary>
+/// Represents a Named Pipe endpoint.
+/// </summary>
+public sealed class NamedPipeEndPoint : EndPoint
+{
+    internal const string LocalComputerServerName = ".";
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="NamedPipeEndPoint"/> class.
+    /// </summary>
+    /// <param name="pipeName">The name of the pipe.</param>
+    public NamedPipeEndPoint(string pipeName) : this(pipeName, LocalComputerServerName)
+    {
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="NamedPipeEndPoint"/> class.
+    /// </summary>
+    /// <param name="pipeName">The name of the pipe.</param>
+    /// <param name="serverName">The name of the remote computer to connect to, or "." to specify the local computer.</param>
+    public NamedPipeEndPoint(string pipeName, string serverName)
+    {
```

Comment:
This dovetails into the conversation going on in the thread model. Do you want to support connecting to other hosts? Is allowing a serverName intended to make this api represent the concept of a NamedPipe endpoint and as it has a server name, you're enabling representing it? Or is the intention to allow cross machine communication?

Reply:
I just followed the flow of this object through and I'm convinced that for a service side application, including serverName doesn't make any sense. You only pass the path to NamedPipeServerStream as you can't create a named pipe on another host. NamedPipeServerStream inserts `\\.\pipe\` in front of the pipe name that you pass.

Reply:
We added the server name here if someone wants to use this endpoint in a client. When Kestrel starts up, it validates that the server name is `.`.